prow/spyglass/lenses/podinfo/podinfo.go

```diff
@@ -98,22 +141,36 @@ func (lens Lens) Body(artifacts []api.Artifact, resourceDir string, data string,
 		return fmt.Sprintf("Failed to load template file: %v", err)
 	}
 
-	p := k8sreporter.PodReport{}
-	if err := json.Unmarshal(content, &p); err != nil {
-		logrus.WithError(err).Infof("Error unmarshalling PodReport")
-		return fmt.Sprintf("Couldn't unmarshal podinfo.json: %v", err)
+	var podLink string
+	if len(conf.RunnerConfigs) > 0 && p.Pod.Name != "" && pj.Spec.Cluster != "" {
```

Comment:
nit: There is no need to check the number of `RunnerConfigs`.

Reply:
this is mainly for covering the case of nil map, which would panic at runtime

Reply:
Reading from nil maps is actually safe and even idiomatic in Go. Panics only occur when writing to a nil map. ([e.g.](https://go.dev/play/p/vRfU9dWMSmE))
This doesn't really matter though.